src/app/qgsapplayertreeviewmenuprovider.cpp

```diff
@@ -836,6 +835,29 @@ QMenu *QgsAppLayerTreeViewMenuProvider::createContextMenu()
             }
           }
         } );
+
+        QAction *showMatchingInAttributeTable = new QAction( QgsApplication::getThemeIcon( QStringLiteral( "/propertyicons/attributes.svg" ) ), tr( "Show in Attribute Table" ), menu );
+        menu->addAction( showMatchingInAttributeTable );
+        connect( showMatchingInAttributeTable, &QAction::triggered, this, [layerId, ruleKey ]
+        {
+          if ( QgsVectorLayer *layer = qobject_cast< QgsVectorLayer * >( QgsProject::instance()->mapLayer( layerId ) ) )
+          {
+            bool ok = false;
+            QString filterExp = layer->renderer() ? layer->renderer()->legendKeyToExpression( ruleKey, layer, ok ) : QString();
+            if ( ok )
+            {
+              const QString canvasFilter = QgsMapCanvasUtils::filterForLayer( QgisApp::instance()->mapCanvas(), layer );
+              if ( canvasFilter == QLatin1String( "FALSE" ) )
+                return;
+              else if ( !canvasFilter.isEmpty() )
+                filterExp = QStringLiteral( "(%1) AND (%2)" ).arg( filterExp, canvasFilter );
```

Comment:
This behavior is a bit strange and kind of unexpected...
So far, the temporal controller plays no role in the attribute table's visibility of features (this is how it is -- I'm not saying it's good or bad), yet with this feature the table will only contain temporally filtered features.
Furthermore, if all features are filtered out (`FALSE` case) then I'd expect an attribute table with a filter that returns no features instead of no attribute table at all!

Reply:
ping @nirvn @nyalldawson